scripts/packages/VSCodeServer/src/trees.jl

```diff
@@ -80,28 +80,50 @@ end
 
 getfield_safe(x, f, default=UNDEF) = isdefined(x, f) ? getfield(x, f) : default
 
-function treerender(x)
+struct PropertyBox
+    object
+end
+treerender(x::PropertyBox) = invoke(treerender, Tuple{Any,String,String}, x.object, "#properties", wsicon(x))
+
+function treerender(x, typedisplay=typeof(x), icon=wsicon(x))
     fields = fieldnames(typeof(x))
 
     if isempty(fields)
-        treerender(Text(string(typeof(x), "()")))
+        treerender(Text(string(typedisplay, "()")))
     else
-        treerender(LazyTree(string(typeof(x)), wsicon(x), function ()
-            collect([SubTree(string(f), wsicon(getfield_safe(x, f)), getfield_safe(x, f)) for f in fields])
```

Comment:
I removed collect because it makes a shallow copy of an already allocated array that seems to be of no use here.

Reply:
We do actually need the `collect`s  here:
```
julia> x = @SArray [1,2,3]
3-element SArray{Tuple{3},Int64,1,3} with indices SOneTo(3):
 1
 2
 3

julia> [x^2 for (i,x) in zip(keys(x), vec(x))]
3-element SizedArray{Tuple{3},Int64,1,1} with indices SOneTo(3):
 1
 4
 9

julia> collect([x^2 for (i,x) in zip(keys(x), vec(x))])
3-element Array{Int64,1}:
 1
 4
 9
 ```
 I've added a few comments about why it's necessary.

Reply:
OMG😯! I was not aware of that occasional iterator type dependency of comprehensions. But makes sense if special indexing is involved.